caikit/core/data_model/base.py

```diff
@@ -235,7 +237,7 @@ def parse_proto_descriptor(mcs, cls):
         _DataBaseMetaClass.class_registry[cls.full_name] = cls
 
         # Add properties that use the underlying backend
-        for field in cls.fields:
+        for field in cls.fields + tuple(old_fields):
             setattr(cls, field, mcs._make_property_getter(field))
```

Comment:
unless we do this, the `foo1.data_stream` is None.

For BazObj, `cls.field` only has `foo` and `bar` ,  which means whenever we want to get those 2 fields, the `_name` is used. But this isn’t done for `datastream`, hence when we try to fetch `foo1.data_stream`, it doesn’t fetch the `_datastream value` (which we set in the `__init__` function)